pkg/applicationserver/applicationserver.go

```diff
@@ -610,6 +816,7 @@ func (as *ApplicationServer) handleUplink(ctx context.Context, ids ttnpb.EndDevi
 					if err != nil {
 						return nil, nil, errFetchAppSKey.WithCause(err)
 					}
+					previousSession := dev.Session
```

Comment:
This will cause a `panic` in `recalculateDownlinkQueue` if `nil`
https://github.com/TheThingsNetwork/lorawan-stack/pull/2163/files#diff-2719373b9eca80ef1abcf09309389399R715 https://github.com/TheThingsNetwork/lorawan-stack/pull/2163/files#diff-2719373b9eca80ef1abcf09309389399R725